src/components/DisplayArticlesData.jsx

```diff
@@ -1,60 +1,43 @@
-import React, { useState, useEffect } from 'react';
-import { getArticleData } from '../modules/articlesData';
+import React, { useEffect } from 'react';
+import { getArticleData, getSpecificArticle } from '../modules/articlesData';
+import ArticleCard from './ArticleCard';
+import SingleArticle from './SingleArticle';
+import { useSelector, useDispatch } from 'react-redux';
 import { Card } from 'semantic-ui-react';
 
 const DisplayArticlesData = () => {
-  const [articlesData, setArticlesData] = useState([]);
+  const dispatch = useDispatch();
+  const articles = useSelector((state) => state.articlesIndex);
+  const singleArticle = useSelector((state) => state.singleArticle);
+
   const getArticles = async () => {
-    let result = await getArticleData();
-    setArticlesData(result.data.articles);
+    const result = await getArticleData();
+    dispatch({ type: 'SET_ARTICLES_INDEX', payload: result.data.articles });
+    dispatch({ type: 'SET_SINGLE_ARTICLE' });
```

Comment:
why are you dispatching for a single article here?
I would either pass dispatch as props to the module here, so you can dispatch it directly that way, or do as Thomas showed you yesterday to refactor the store so you can access it from a module. 
In that case you would make the `getArticleData` global to use from anywhere and you don't have to pass that function around but can simply import it where you need it

Reply:
We refactored that. Thanks.